mne/viz/evoked.py

```diff
@@ -1698,258 +1636,215 @@ def _evoked_sensor_legend(info, picks, ymin, ymax, show_sensors, ax):
-    ci_dict, data_dict = dict(), dict()
-
-    for cond in conditions:
-        this_evokeds = evokeds[cond]
-        # this will fail if evokeds do not have the same structure
-        # (e.g. channel count)
-        res = _get_data_and_ci(this_evokeds, scaling=scaling, picks=picks,
-                               ci_fun=ci_fun, gfp=gfp)
-        data_dict[cond] = res[0]
-        if ci_fun is not None:
-            ci_dict[cond] = res[1]
-    return data_dict, ci_dict
+    return (data,) if ci_fun is None else (data, ci)
 
 
-def _get_ci_function_for_evokeds(ci):
-    """Get the function for calculating the confidence interval for evokeds."""
-    # check ci parameter
+def _get_ci_function_pce(ci):
+    """Get confidence interval function for plot_compare_evokeds."""
     if ci is None:
         return None
-
-    if ci is True:
-        ci = .95
-    elif ci is not False and not (isinstance(ci, np.float) or callable(ci)):
-        raise TypeError('ci must be None, bool, float or callable, got %s' %
-                        type(ci))
-
-    _ci_fun = None
-    if ci is not False:
-        if callable(ci):
-            _ci_fun = ci
-        else:
-            from ..stats import _ci
-            _ci_fun = partial(_ci, ci=ci, method="bootstrap")
-    return _ci_fun
-
-
-def _finish_styles_plot_comp_evoked(styles, colors, linestyles, conditions,
-                                    cmap):
-    """Finalize styles for plot_compare_evokeds."""
-    # Styles (especially color and linestyle) are pulled from a dict 'styles'.
-    # This dict has one entry per condition. Its color and linestyle entries
-    # are pulled from the 'colors' and 'linestyles' dicts via '/'-tag matching
-    # unless they are overwritten by entries from a user-provided 'styles'.
-
-    # check if input is valid
-    if isinstance(styles, dict):
-        for style_ in styles:
-            if style_ not in conditions:
-                raise ValueError("Could not map between 'styles' and "
-                                 "conditions. Condition " + style_ +
-                                 " was not found in the supplied data.")
-
-    # color
-    if (colors is not None and not isinstance(colors, str) and
-            not isinstance(colors, dict) and len(colors) > 1):
-        colors = {condition: color for condition, color
-                  in zip(conditions, colors)}
-
-    cmap_label = ""
-    if cmap is not None:
-        if not isinstance(cmap, str) and len(cmap) == 2:
-            cmap_label, cmap = cmap
-
-    styles, the_colors, color_conds, color_order, colors_are_float =\
-        _setup_styles(conditions, styles, cmap, colors, linestyles)
-
-    return (styles, colors, the_colors, color_conds, color_order,
-            colors_are_float, cmap_label, cmap)
+    elif callable(ci):
+        return ci
+    elif isinstance(ci, bool) and not ci:
+        return None
+    elif isinstance(ci, bool):
+        ci = 0.95
```

Comment:
```
elif isintance(ci, bool):
    if ci:
        return .95
    else:
        return None
```

Reply:
we don't want to return `0.95` here.  we want to change it from `True` to `0.95` so it can go through the call `partial()` a few lines down.